app/src/org/commcare/dalvik/dialogs/CustomProgressDialog.java

```diff
@@ -200,17 +196,37 @@ public void onClick(View v) {
         }
     }
 
-    private void setupCancelButton(View v) {
+    private Button setupCancelButton(View v) {
         Button b = (Button)v.findViewById(R.id.dialog_cancel_button);
         b.setOnClickListener(new OnClickListener() {
 
             @Override
             public void onClick(View v) {
                 ((CommCareActivity)getActivity()).cancelCurrentTask();
+                showCancelledState();
             }
 
         });
         b.setVisibility(View.VISIBLE);
+        return b;
+    }
+
+    private void showCancelledState() {
+        wasCancelPressed = true;
+
+        AlertDialog pd = (AlertDialog)getDialog();
+        if (pd != null) {
+            TextView titleView = (TextView)pd.findViewById(R.id.progress_dialog_title).findViewById(R.id.dialog_title_text);
+            TextView messageView = (TextView)pd.findViewById(R.id.progress_dialog_message);
+            Button cancelButton = (Button)pd.findViewById(R.id.dialog_cancel_button);
+            setCancellingText(titleView, messageView, cancelButton);
+        }
+    }
+
+    private void setCancellingText(TextView titleTextView, TextView messageTextView, Button cancelButton) {
+        titleTextView.setText(cancellingPrefixText + " '" + title + "'");
+        messageTextView.setText(cancellingPrefixText + "...");
```

Comment:
This kind of verbage is full of some dangerous assumptions about the nature of the language involved. If we were in a right-to-left language this would be really confusing to put together.

Two options that might help:

1) Use the localization keys to do the construction, IE: 

Localization.get("cancelling.title", new String[] {titleText})
where
cancelling.title=Cancelling: ${0}

or something similar. 

Alternatively, you could only use the message view for the cancel message (although I'd put the "..." in the message itself for the same reason)